### PR TITLE
Stream aliases

### PR DESCRIPTION
**Pull request by jlashner**
Here's a first attempt at adding aliases into the HK data stream. The alias file can be set in the `hub` section of your site-config file, or via the command line with the `--alias-file` argument. My hub looks like this: 
```
hub:

  wamp_server: ws://localhost:8001/ws
  wamp_http: http://localhost:8001/call
  wamp_realm: test_realm
  address_root: observatory
  registry_address: observatory.registry
  alias_file: ${OCS_CONFIG_DIR}/aliases.yaml
```
which works from both within docker and on the host. 

The alias file is a yaml file with entries structured like:
```
instance-id:
    feed-name:
        stream-1: "alias-1"
        stream-2: ["alias-2.1", "alias-2.2"]
```
where you can give a single alias or list of aliases for each stream. For example, for a lakeshore240 agent you might have something like this
```
LSASIM:
  temperatures:
    "Channel 01 V": ["CHWP1_V", "D001_V"]
    "Channel 01 T": ["CHWP1_T", "D001_T"]
    "Channel 02 V": ["CHWP2_V", "D002_V"]
    "Channel 02 T": ["CHWP2_T", "D002_T"]
    "Channel 03 V": ["CHWP3_V", "D003_V"]
    "Channel 03 T": ["CHWP3_T", "D003_T"]
    "Channel 04 V": ["CHWP4_V", "D004_V"]
    "Channel 04 T": ["CHWP4_T", "D004_T"]
    "Channel 05 V": ["CHWP5_V", "D005_V"]
    "Channel 05 T": ["CHWP5_T", "D005_T"]
    "Channel 06 V": ["CHWP6_V", "D006_V"]
    "Channel 06 T": ["CHWP6_T", "D006_T"]
    "Channel 07 V": ["CHWP7_V", "D007_V"]
    "Channel 07 T": ["CHWP7_T", "D007_T"]
    "Channel 08 V": ["CHWP8_V", "D008_V"]
    "Channel 08 T": ["CHWP8_T", "D008_T"]
```

These entries are read in and saved by each OCS Feed when it's registered. The aliases for each provider are written as a G3MapVectorString in the hk status frame, so each provider object looks like:
```
{  'aliases': <spt3g.core.G3MapVectorString object at 0x7fd68074c6b8>,
   'description': <spt3g.core.G3String object at 0x7fd68074c660>,
   'prov_id': <spt3g.core.G3Int object at 0x7fd68074c978>
}
```
The only thing that I'm unsure about is whether or not we want to store the aliases in the so3g hksession providers, so that they're written to the status frame automatically. Currently, the aggregator is adding them manually to the status frame after the hksession creates it.

### CODE REVIEW

File: agents/aggregator/aggregator_agent.py [hunk 1]
```diff
@@ -32,13 +33,16 @@ class Provider:
             Start time of current frame
         blocks (dict):
             All blocks that are written by provider.
+        aliases (dict):
+            Alias map from stream name to list of aliases
```
Comment (mhasself) [inline agents/aggregator/aggregator_agent.py]: Is "stream name" the official name for a uniquely named vector of data in a feed?  We have been using "field" in sisock.  I would prefer to use field or channel, generally.

File: ocs/ocs_agent.py [hunk 2]
```diff
@@ -98,6 +98,7 @@ def __init__(self, config, site_args, address=None):
         self.next_session_id = 0
         self.session_archive = {} # by op_name, lists of OpSession.
         self.agent_address = address
+        self.root_address, self.instance_id = self.agent_address.split('.')
```
[the remark this answers is not in the document]
Comment (jlashner) [reply ocs/ocs_agent.py]: I think the aggregator assumes the agent address is of the form `<root>.<instance_id>`. Will this not always be the case? I think it's a good idea to not allow '.' in the root or instance names.

[the remark this answers is not in the document]
Comment (jlashner) [reply ocs/ocs_agent.py]: Will there be cases where we want more segments before the instance id?

File: agents/aggregator/aggregator_agent.py (hunk 1 above)
Comment (jlashner) [reply agents/aggregator/aggregator_agent.py]: I'm ok switching it to `field`.

**Comment by jlashner** [comment]
Closed because we decided to just make sure each institution is recording where they are placing each thermometer for each cooldown

**Pull request closed without merging**